Connection to existing MQTT broker

When executing rv-homekit on a machine that already has MQTT running, I get the following error:

```
pi@control:~/rv-homekit $ ./rv-homekit
2021/05/10 10:55:08 /home/pi/rv-homekit
2021/05/10 10:55:08 No config file found. Making new IDs
2021/05/10 10:55:08 Invalid config file format. Starting new.
2021/05/10 10:55:08 Found 0 tank sensors.
2021/05/10 10:55:08 Registered 0 undiscovered devices.
2021/05/10 10:55:08 Connecting to tcp://127.0.0.1:1883
panic: not Authorized

goroutine 36 [running]:
github.com/jgulick48/rv-homekit/internal/mqtt.(*client).Connect(0x2a16370)
        /home/pi/rv-homekit/build/rv-homekit/internal/mqtt/client.go:68 +0x45c
github.com/jgulick48/rv-homekit/internal/rvhomekit.(*client).registerBatteryLevel.func1(0x4f7d24, 0x2a16370)
        /home/pi/rv-homekit/build/rv-homekit/internal/rvhomekit/client.go:194 +0x24
created by github.com/jgulick48/rv-homekit/internal/rvhomekit.(*client).registerBatteryLevel
        /home/pi/rv-homekit/build/rv-homekit/internal/rvhomekit/client.go:192 +0x370

```
My app is installed in /home/pi/rv-homekit:
```
pi@control:~/rv-homekit $ ls -al
total 10412
drwxr-xr-x  3 pi pi     4096 May 10 02:30 .
drwxr-xr-x 24 pi pi     4096 May 10 10:24 ..
drwxr-xr-x  3 pi pi     4096 May 10 02:32 build
-rw-r--r--  1 pi pi      114 May 10 02:25 config.json
-rwxr-xr-x  1 pi pi 10644855 May 10 03:18 rv-homekit
pi@control:~/rv-homekit $ pwd
/home/pi/rv-homekit
pi@control:~/rv-homekit $ more config.json
{
  "bridgeName": "My RV",
  "openHabServer": "http://192.168.1.4:8080",
  "debug": true,
  "mqttConfiguration": {
    "host": "192.168.1.10",
    "port": 1883,
    "deviceId": "rv-homekit",
    "username": "[REDACTED]",
    "password": "[REDACTED]"
  },
  "pin": "19191919",
  "port": "12321"
}
pi@control:~/rv-homekit $
```

How would one direct the connection to an existing MQTT server using a specified username and password?

Thanks!

--- a/internal/models/models.go
+++ b/internal/models/models.go
@@ -29,6 +29,8 @@ type MQTTConfiguration struct {
 	Host     string `json:"host"`
 	Port     int    `json:"port"`
 	DeviceID string `json:"deviceId"`
+	Username string `json:"username"`
+	Password string `json:"password"`
 }
 
 type BMVConfig struct {

--- a/internal/mqtt/client.go
+++ b/internal/mqtt/client.go
@@ -75,6 +75,10 @@ func (c *client) Connect() {
 	opts.AddBroker(fmt.Sprintf("tcp://%s:%d", c.config.Host, c.config.Port))
 	opts.SetClientID("go_mqtt_client")
 	opts.SetDefaultPublishHandler(c.messagePubHandler)
+	if c.config.Username != "" && c.config.Password != "" {
+		opts.SetUsername(c.config.Username)
+		opts.SetPassword(c.config.Password)
+	}
 	opts.OnConnect = connectHandler
 	opts.OnConnectionLost = c.connectLostHandler
 	c.mqttClient = mqtt.NewClient(opts)

